src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java

```diff
@@ -155,33 +155,43 @@ public synchronized void addRow(Object... values) throws SQLServerException {
      */
     private void internalAddrow(JDBCType jdbcType, Object val, Object[] rowValues,
             Map.Entry<Integer, SQLServerDataColumn> pair) throws SQLServerException {
+        if (null != val) {
+            SQLServerDataColumn currentColumnMetadata = pair.getValue();
+            int nValueLen;
+            switch (jdbcType) {
+                case BIGINT:
+                    rowValues[pair.getKey()] = (val instanceof Long) ? val : Long.parseLong(val.toString());
+                    break;
+
+                case BIT:
+                    if (val instanceof Boolean) {
+                        rowValues[pair.getKey()] = val;
+                    } else if (val.toString().equals("0")
+                            || val.toString().equalsIgnoreCase(Boolean.FALSE.toString())) {
+                        rowValues[pair.getKey()] = Boolean.FALSE;
+                    } else if (val.toString().equals("1") || val.toString().equalsIgnoreCase(Boolean.TRUE.toString())) {
```

Comment:
Create a local String variable for `val.toString()` so we don't end up calling it 4 times.